tests/components/dyson/test_climate.py

```diff
@@ -154,236 +136,259 @@ def _get_device_heat_on():
+    await hass.async_add_executor_job(update_callback, MockDysonState())
+    await hass.async_block_till_done()
+
+    state = hass.states.get("climate.temp_name")
+    assert state.attributes[ATTR_FAN_MODE] == FAN_DIFFUSE
+
+    device.state.heat_mode = HeatMode.HEAT_ON.value
+    device.state.heat_state = HeatState.HEAT_STATE_OFF.value
+    await hass.async_add_executor_job(update_callback, MockDysonState())
+    await hass.async_block_till_done()
+
+    state = hass.states.get("climate.temp_name")
+    assert state.state == HVAC_MODE_HEAT
+    assert state.attributes[ATTR_HVAC_ACTION] == CURRENT_HVAC_IDLE
+
+    device.environmental_state.humidity = 0
+    await hass.async_add_executor_job(update_callback, MockDysonState())
+    await hass.async_block_till_done()
+
+    state = hass.states.get("climate.temp_name")
+    assert state.attributes.get(ATTR_CURRENT_HUMIDITY) is None
+
+    device.environmental_state = None
+    await hass.async_add_executor_job(update_callback, MockDysonState())
+    await hass.async_block_till_done()
+
+    state = hass.states.get("climate.temp_name")
+    assert state.attributes.get(ATTR_CURRENT_HUMIDITY) is None
+
+    device.state.heat_mode = HeatMode.HEAT_OFF.value
+    device.state.heat_state = HeatState.HEAT_STATE_OFF.value
+    await hass.async_add_executor_job(update_callback, MockDysonState())
+    await hass.async_block_till_done()
+
+    state = hass.states.get("climate.temp_name")
+    assert state.state == HVAC_MODE_COOL
+    assert state.attributes[ATTR_HVAC_ACTION] == CURRENT_HVAC_COOL
+
+
+@patch(
+    "homeassistant.components.dyson.DysonAccount.devices",
+    return_value=[],
+)
+@patch("homeassistant.components.dyson.DysonAccount.login", return_value=True)
+async def test_setup_component_without_devices(mocked_login, mocked_devices, hass):
+    """Test setup component with no devices."""
+    await async_setup_component(hass, DYSON_DOMAIN, _get_config())
+    await hass.async_block_till_done()
+
+    entity_ids = hass.states.async_entity_ids(DOMAIN)
+    assert not entity_ids
+
+
+@patch(
+    "homeassistant.components.dyson.DysonAccount.devices",
+    return_value=[_get_device_heat_on()],
+)
+@patch("homeassistant.components.dyson.DysonAccount.login", return_value=True)
+async def test_dyson_set_temperature(mocked_login, mocked_devices, hass):
+    """Test set climate temperature."""
+    await async_setup_component(hass, DYSON_DOMAIN, _get_config())
+    await hass.async_block_till_done()
+
+    device = mocked_devices.return_value[0]
+    device.temp_unit = TEMP_CELSIUS
+
+    # Without correct target temp.
+    await hass.services.async_call(
+        DOMAIN,
+        SERVICE_SET_TEMPERATURE,
+        {
+            ATTR_ENTITY_ID: "climate.temp_name",
+            ATTR_TARGET_TEMP_HIGH: 25.0,
+            ATTR_TARGET_TEMP_LOW: 15.0,
+        },
+        True,
+    )
+
+    set_config = device.set_configuration
+    assert set_config.call_count == 0
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=6184085)